### DISCUSSION TITLE
Message: an outstanding start workflow request is in-progress. Failed to acquire the resource

### DISCUSSION THREAD

**Discussion by alexfjw** [uber/cadence#4439] [Q&A] [answered]
Hi team, we encountered a small increase in this error in our history instances for a short time (~20s) with a sharp increase in latency and errors in starting new workflows for that period. We did not see any issues in our db or infra. 

Any idea what this may be caused by or what we can do to prevent this error? (server version 0.21.3)

```
Fail to process task

ServiceBusyError{Message: an outstanding start workflow request is in-progress. Failed to acquire the resource.}
```

**Comment by longquanzheng** [comment, marked as answer]
The error is from here: https://github.com/uber/cadence/blob/v0.21.3/service/history/historyEngine.go#L566

From the code, it's very likely that you are trying to start more than one workflows with the same workflowID concurrently.  

If that's the case, those requests will be all trying to acquire a lock for the worklfowID which would cause some failed.
